src/test/java/proj/kedabra/billsnap/presentation/controllers/GetBillControllerIT.java

```diff
@@ -0,0 +1,202 @@
+package proj.kedabra.billsnap.presentation.controllers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.get;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.List;
+
+import javax.transaction.Transactional;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import proj.kedabra.billsnap.business.utils.enums.BillStatusEnum;
+import proj.kedabra.billsnap.fixtures.BillCreationResourceFixture;
+import proj.kedabra.billsnap.fixtures.UserFixture;
+import proj.kedabra.billsnap.presentation.ApiError;
+import proj.kedabra.billsnap.presentation.resources.BillResource;
+import proj.kedabra.billsnap.presentation.resources.BillSplitResource;
+import proj.kedabra.billsnap.presentation.resources.ShortBillResource;
+import proj.kedabra.billsnap.security.JwtService;
+import proj.kedabra.billsnap.utils.SpringProfiles;
+
+@Tag("integration")
+@ActiveProfiles(SpringProfiles.TEST)
+@SpringBootTest
+@AutoConfigureMockMvc
+@SuppressWarnings("squid:S00112")
+@AutoConfigureTestDatabase
```

Comment:
Do we need this

Reply:
dont think so, will remove it

Reply:
Tested without the annotation and tests fails, so I'll keep it there